File: javascript/1133-largest-unique-number.js
/**
 * 1133 - Largest Unique Number
 * Hashmap | Count frequencies
 * Time: O(n) | Space: O(n)
 * @param {number[]} nums
 * @return {number}
 */
const largestUniqueNumber = function (nums) {
  const counts = new Map();
  let res = -1;

  for (const num of nums) {
    counts.set(num, (counts.get(num) || 0) + 1);
  }

  for (const [num, count] of counts) {
    if (count === 1) {
      res = Math.max(res, num);
    }
  }

  return res;
};

File: python/1133-largest-unique-number.py
"""
1133 - Largest Unique Number [Easy]
Hash Table | Counting | Array
Time: O(n) | Space: O(n)

Given an integer array nums, return the largest integer that only occurs once. If no integer occurs once, return -1.
"""

from collections import defaultdict
from typing import List


class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        res = -1

        for num in nums:
            counts[num] += 1

        for num, count in counts.items():
            if count == 1:
                res = max(res, num)

        return res
